### DISCUSSION TITLE
Scalafmt does not preserve line breaks in single line nested for and if

### DISCUSSION THREAD

**Discussion by vicaba** [scalameta/scalafmt#3553] [Q&A] [answered]
Hello,

I am trying to set up `scalafmt` scala 3 project.

Somewhere in the code, inside a `def` I have:
```scala
for row <- zeroCells.rows.indices do 
  if zeroCells.rows(row) == 1 then 
    for col <- m(row).indices do 
      m(row)(col) = 0
```

However, when I format with `scalafmt`, it is rewritten as:

```scala
for row <- zeroCells.rows.indices do if zeroCells.rows(row) == 1 then for col <- m(row).indices do m(row)(col) = 0
```

which is hard to read. I'd like to preserve this sort of line break and indentation (the original snippet I've shown). Scalafmt does seem to format the following snippet as I expect (see, however, that I have added two `println` to transform the blocks inside to multiline):
```scala 3
for col <- zeroCells.cols.indices do
  if zeroCells.cols(col) == 1 then
    println("a") // This was added
    for row <- m.indices do
      println("b") // This was added
      m(row)(col) = 0
```

I have seen this similar issue from 2016: https://github.com/scalameta/scalafmt/issues/366, but most of the links are broken, and I can no longer find the directive
```
keepSelectChainLineBreaks = true
```
in the documentation for scalafmt 3.7.4. It also may be that what I am looking for is not related to the issue https://github.com/scalameta/scalafmt/issues/366, since my issue is not really related with _chains_.

Is there a directive I am missing to achieve the expected behavior?

Thank you.

**Comment by kitbellew** [comment, marked as answer]
@vicaba rather than searching through old issues, I'd recommend re-reading the documentation. One option is: https://scalameta.org/scalafmt/docs/configuration.html#newlinessource

**Reply by vicaba** [reply]
Thank you. 
I indeed searched the docs, but missed that part. Then, the only thing I could find was that old issue. My apologies.
I have "fixed" that problem now and others, but still battling with some other formatting directives. I will keep trying for the rest of the week.
